pkg/services/provisioning/values/values_test.go

```diff
@@ -241,6 +246,30 @@ func TestValues(t *testing.T) {
 	})
 }
 
+func TestValues_readFile(t *testing.T) {
+	type Data struct {
+		Val StringValue `yaml:"val"`
+	}
+
+	f, err := ioutil.TempFile(os.TempDir(), "file expansion *")
+	require.NoError(t, err)
+	file := f.Name()
+
+	defer func() {
+		require.NoError(t, os.Remove(file))
```

Comment:
I think it's safer to ensure the file is closed before you try to remove it. Depending on the platform, removing an open file could fail:
```suggestion
		require.NoError(t, f.Close())
		require.NoError(t, os.Remove(file))
```

Reply:
Reverted this, I already close the file immediately after writing to it so I got an error for trying to close a file twice